### PR TITLE
Add trace module that uses GCP CloudTrace API to register spans.

### PR DESCRIPTION
**Pull request by TyeMcQueen**

### CODE REVIEW

File: trace/trace.go [hunk 1]
```diff
@@ -0,0 +1,486 @@
+// This module makes it easy to register a Span (and associated Trace)
```
Comment (jonnydgreen) [inline trace/trace.go:1]: Will you be writing integration/unit tests for this in the future?

File: trace/trace.go [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+			return nil, nil, err
+		} else {
+			project = dflt
+		}
+	}
+	prefix := "project/" + project + "/"
+	for ; 0 < runners; runners-- {
+		go func() {
+			for sp := range queue {
+				// TODO!  Add a reasonable timeout!!
```
Comment (jonnydgreen) [inline trace/trace.go:235]: I presume in subsequent PRs?

**Review by jonnydgreen — APPROVED** [review]
LGTM! Thanks for the docstrings on each of the functions, was really helpful!

File: trace/trace.go (hunk 1 above)
Comment (TyeMcQueen) [reply trace/trace.go:1]: Yes!

File: trace/trace.go (hunk 2 above)
Comment (TyeMcQueen) [reply trace/trace.go:235]: Yes.  Also need to make it more resilient in a couple of other ways.  Trying to take smaller bites at first. :)

**Pull request merged**